apps/dashboard/app/models/configuration_property.rb

```diff
@@ -0,0 +1,61 @@
+class ConfigurationProperty
+
+  def self.with_boolean_mapper(name:, default_value: nil, read_from_env: false, env_name: nil)
+    ConfigurationProperty.new(name, default_value, read_from_env, env_name, BooleanMapper)
+  end
+
+  def self.with_integer_mapper(name:, default_value: nil, read_from_env: false, env_name: nil)
+    ConfigurationProperty.new(name, default_value, read_from_env, env_name, IntegerMapper)
+  end
+
+  def self.property(name:, default_value: nil, read_from_env: false, env_name: nil)
+    ConfigurationProperty.new(name, default_value, read_from_env, env_name, PassThroughMapper)
+  end
+
+  attr_reader :name, :default_value, :read_from_environment, :environment_name
+  alias_method :read_from_environment?, :read_from_environment
+
+  def initialize(name, default_value, read_from_env, env_name, mapper)
+    @name = name.to_sym
+    @default_value = default_value
+    @read_from_environment = read_from_env
+    @mapper = mapper
+
+    environment_name = env_name || "OOD_#{@name.to_s.upcase}"
+    @environment_name = environment_name if @read_from_environment
+  end
+
+  def map_string(value)
+    @mapper.map_string(value)
+  end
+
+  class PassThroughMapper
+    def self.map_string(string_value)
+      string_value
+    end
+  end
+
+  class IntegerMapper
+    def self.map_string(string_value)
+      string_value.nil? ? string_value : Integer(string_value.to_s)
```

Comment:
I think this is the behavior we want - I'm just leaving this comment in case anyone in the future questions it.

```
irb(main):001:0> "10.invalid".to_i
=> 10
irb(main):002:0> Integer("10.invalid")
Traceback (most recent call last):
        5: from /home/jeff/apps/ruby/2.7.6/bin/irb:23:in `<main>'
        4: from /home/jeff/apps/ruby/2.7.6/bin/irb:23:in `load'
        3: from /home/jeff/apps/ruby/2.7.6/lib/ruby/gems/2.7.0/gems/irb-1.2.6/exe/irb:11:in `<top (required)>'
        2: from (irb):2
        1: from (irb):2:in `Integer'
ArgumentError (invalid value for Integer(): "10.invalid")
irb(main):003:0> "invalid".to_i
=> 0
```